code/components/gta-net-five/src/CloneObjectManager.cpp

```diff
@@ -96,6 +110,25 @@ static void netObjectMgrBase__ChangeOwner(rage::netObjectMgr* manager, rage::net
 	object->PostMigrate(migrationType);
 
 	CloneObjectMgr->ChangeOwner(object, oldOwnerId, targetPlayer, migrationType);
+	// Handle scenarios where a train was previously owned by the server
+#ifdef GTA_FIVE
+	// Make sure that this is a train and that we are now the new owner of it
+	if (object->objectType == (uint16_t)NetObjEntityType::Train && targetPlayer->physicalPlayerIndex() == rage::GetLocalPlayer()->physicalPlayerIndex())
+	{
+		// Ensure that the vehicle isn't a nullptr
+		if (CVehicle* train = (CVehicle*)object->GetGameObject())
+		{
+			// Ensure this is the engine and the client has no knowledge of the trains current track node.
+			if (CTrain__IsCarriageEngine(train) && *(int*)((char*)train + TrainTrackNodeIndexOffset) == 0)
+			{
+				//Find the trains track node based on its current location
+				CTrain__SetTrainCoord(train, -1, -1);
+				// Force blend to apply location
+				object->GetBlender()->m_30();
```

Comment:
Nit: we can also rename `m_30()` to `Update()` in `netBlender`. So there is more context on what the function we are calling is about.